pkg/assetfetcher/http_test.go

```diff
@@ -633,6 +632,53 @@ func TestMakeRequestSuccess(t *testing.T) {
 	assert.NotNil(t, <-assetChan)
 }
 
+func TestMakeRequestSkipAndStatUnscannedAssets(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockRT := NewMockRoundTripper(ctrl)
+	mockStatFn := NewMockStat(ctrl)
+
+	assetScanned := Asset{
+		IP:      "127.0.0.1",
+		ID:      1,
+		History: assetHistoryEvents{AssetHistory{Type: "SCAN", Date: "2019-06-14T15:03:47.000Z"}}}
+	assetUnscanned := Asset{
+		IP:      "127.0.0.2",
+		ID:      2,
+		History: assetHistoryEvents{AssetHistory{Type: "ASSET-IMPORT", Date: "2019-06-14T15:03:47.000Z"}}}
+	resp := SiteAssetsResponse{
+		Resources: []Asset{assetScanned, assetUnscanned},
+		Page:      Page{},
+	}
+
+	respJSON, _ := json.Marshal(resp)
+	mockRT.EXPECT().RoundTrip(gomock.Any()).Return(&http.Response{
+		Body:       ioutil.NopCloser(bytes.NewReader(respJSON)),
+		StatusCode: http.StatusOK,
+	}, nil)
+
+	host, _ := url.Parse("http://localhost")
+	nexposeAssetFetcher := &NexposeAssetFetcher{
+		HTTPClient: &http.Client{Transport: mockRT},
+		Host:       host,
+		PageSize:   100,
+		StatFn:     func(ctx context.Context) domain.Stat { return mockStatFn },
+	}
+	mockStatFn.EXPECT().Count("assetreceived.skipped", float64(1)).Times(1)
+
+	var wg sync.WaitGroup
+	assetChan := make(chan domain.AssetEvent, 1)
+	errChan := make(chan error, 1)
+	defer close(assetChan)
+	defer close(errChan)
+
+	wg.Add(1)
```

Comment:
task: I know you intentionally removed the `wg.Wait()` call because it wasn't strictly needed for the test. However, I believe it actually does provide some testing value. Notably, the `makeRequest` method is obligated to call `wg.Done()` in order to signal the it has completed its work. That implies it is a bug if the method never calls `wg.Done()` and it should fail the test if that is the case.

There's direct way to check how many outstanding tasks there are in a `WaitGroup`, only the ability to add one, remove one, and wait for all to finish. Given that, I recommend bring the `wg.Wait()` back since it will guarantee that the method calls `Done()`. If the method does not then the test will fail due to timeout.

Reply:
can do! and i'll add wg.Wait() to the other tests too

Reply:
fixed!